README example failing to compile with `-std=c++23`

Godbolt: https://godbolt.org/z/GWKc4Kfax
(basically just the example godbolt link, then change the std to `c++23`)

```
In file included from /opt/compiler-explorer/libs/stdexec/trunk/include/stdexec/__detail/__execution_fwd.hpp:18,
                 from /opt/compiler-explorer/libs/stdexec/trunk/include/stdexec/execution.hpp:18,
                 from <source>:1:
/opt/compiler-explorer/libs/stdexec/trunk/include/stdexec/__detail/__config.hpp: In function 'void stdexec::__terminate()':
/opt/compiler-explorer/libs/stdexec/trunk/include/stdexec/__detail/__config.hpp:571:5: error: 'unreachable' is not a member of 'std'
  571 |     STDEXEC_UNREACHABLE();
      |     ^~~~~~~~~~~~~~~~~~~
```

Apparently missing `#include <ultility>`.

--- a/include/stdexec/__detail/__config.hpp
+++ b/include/stdexec/__detail/__config.hpp
@@ -40,6 +40,7 @@
 #include <cassert>
 #include <cstdlib>
 #include <type_traits> // IWYU pragma: keep
+#include <utility> // IWYU pragma: keep for std::unreachable
 
 // When used with no arguments, these macros expand to 1 if the current
 // compiler corresponds to the macro name; 0, otherwise. When used with arguments,

